tests/testthat/test-renderer2-widerect.R

```diff
@@ -1,454 +1,449 @@
+    expect_equal(length(node.set), 0)
+  }
+})
+
+test_that("clicking status legend does not hide text", {
+  node.set <-
+    getNodeSet(html.no.rects,
+               '//g[@class="geom9_text_ts"]//text[@class="geom"]')
+  expect_equal(length(node.set), 1)
+})
+
+clickID("plot_ts_status_variable_even")
+clickID("plot_ts_status_variable_odd")
+html.with.rects <- getHTML()
+
+test_that("clicking status legend brings back tallrects", {
+  for(rect.xpath in rect.xpaths){
+    node.set <- getNodeSet(html.with.rects, rect.xpath)
+    expect_equal(length(node.set), nrow(years))
+    style.list <- list()
+    for(node.i in seq_along(node.set)){
+      node <- node.set[[node.i]]
+      a.vec <- xmlAttrs(node)
+      style.list[[node.i]] <- a.vec[["style"]]
+      sizes <- as.numeric(a.vec[c("height", "width")])
+      expect_true(all(sizes > 0))
+    }
+    style.vec <- do.call(c, style.list)
+    dash.mat <- str_match_perl(style.vec, dasharrayPattern)
+    ## Use paste() to treat NA as a value instead of ignoring it.
+    dash.table <- table(paste(dash.mat[, "value"]))
+    ## There should be 2 unique values of stoke-dasharray.
+    expect_equal(length(dash.table), 2)
+  }
+})
+
+test_that("play restarts animation (second time)", {
+  old.year <- getYear()
+  clickID("play_pause")
+  Sys.sleep(5)
+  new.year <- getYear()
+  expect_true(old.year != new.year)
+})
+
+legend.td.xpath <-
+  '//tr[@class="region_variable"]//td[@class="legend_entry_label"]'
+rects_and_legends <- function(){
+  html <- getHTML()
+  list(rects=getNodeSet(html, '//rect[@id="United_States"]'),
+       legends=getStyleValue(html, legend.td.xpath, "opacity"))
+}
+test_that("clicking legend removes/adds countries", {
+  before <- rects_and_legends()
+  expect_equal(length(before$rects), 1)
+  expect_equal(sum(before$legends=="1"), 14)
+  expect_equal(sum(before$legends=="0.5"), 0)
+  clickID("plot_ts_region_variable_North_America")
+  Sys.sleep(1)
+  oneclick <- rects_and_legends()
+  expect_equal(length(oneclick$rects), 0)
+  expect_equal(sum(oneclick$legends=="1"), 12)
+  expect_equal(sum(oneclick$legends=="0.5"), 2)
+  clickID("plot_ts_region_variable_North_America")
+  Sys.sleep(1)
+  twoclicks <- rects_and_legends()
+  expect_equal(length(twoclicks$rects), 1)
+  expect_equal(sum(twoclicks$legends=="1"), 14)
+  expect_equal(sum(twoclicks$legends=="0.5"), 0)
+})
+
+clickID('updates_ms')
+ 
+
+
+test_that("pause stops animation (third time)", {
+  clickID("play_pause")
+  old.year <- getYear()
+  Sys.sleep(4)
+  new.year <- getYear()
+  expect_true(old.year == new.year)
+})
+
+remDr$Runtime$evaluate("document.getElementsByClassName('show_hide_selector_widgets')[0].dispatchEvent(new CustomEvent('click'));")
+clickSelector("year_variable_selector_widget")
+sendKey("Backspace", "Backspace", 8)
+remDr$Input$insertText(text = "1962")
+clickSelector("year_variable_selector_widget")
+sendKey("ArrowDown", "ArrowDown", 40)
+sendKey("Enter", "Enter", 13)
+   
+ 
+Sys.sleep(3)
```

Comment:
The wait time is increased because the test failed on my laptop with Sys.sleep(1). The test was passing on GitHub actions but failed when running locally